File: cats_and_mouse/cam.js
const absValue = (x) => Math.abs(x);

const catAndMouse = (x, y, z) => {
    const catA = absValue(x - z);
    const catB = absValue(y - z);

    if (catA === catB) {
        return 'Mouse C';
    }else if (catA < catB) {
        return 'Cat A';
    } else {
        return 'Cat B';
    }
}

console.log(catAndMouse(1, 2, 3));
console.log(catAndMouse(1, 3, 2));

File: cats_and_mouse/cam.py
def cat_and_mouse(x:int, y:int, z:int) -> str:
    if abs(x - z) == abs(y - z):
        return 'Mouse C'
    elif abs(x - z) < abs(y - z):
        return 'Cat A'
    else:
        return 'Cat B'


if __name__ == '__main__':
    print(cat_and_mouse(1, 2, 3))
    print(cat_and_mouse(1, 3, 2))